R/Utils.R

```diff
@@ -196,4 +198,74 @@ SeuratToAnnData <- function(seuratObj, outFileBaseName, assayName = NULL, doDiet
   }
 
   return(ret)
+}
+
+#' @title Remove UCell Doublets
+#'
+#' @description Attempts to remove dump-gate "doublets" from the main three cell type subset objects (Tcell, Bcell, MoMacDC)
+#' @param parentUCell The subset's target cell type's UCell score. (Computed by scGate early in processing).
+#' @param otherUCellScores A vector of cell type UCell scores to attempt to discard from downstream processing. By default, these are cell types that comprise the dump gate in RIRA's scGate model. 
+#' @param clusterResolution The metadata column name that defines cluster membership (e.g. ClusterNames_0.2)
+#' @param cosineSimilarityCutoff A number between -1 and 1 that is used to threshold/gate the prospective doublets as defined by the computed score.
+#' @return Returns a seurat object with the UCell doublets removed. 
+#' @export
+#' 
+.RemoveUCellDoublets <- function(seuratObj, parentUCell = NULL, otherUCellScores = c("Platelet_MK_UCell", "Erythrocyte_UCell", "Neutrophil_UCell", "Epithelial_UCell", "Stromal_UCell"), clusterResolution = "ClusterNames_1.2", cosineSimilarityCutoff = 0.5){
```

Comment:
So the software convention is that prefixing with '.' indicates a private method. R is not a very good guide in this world, but if we're exporting (which we should), let's drop the dot

Reply:
sounds good, I'll amend this.